src/libsync/propagateuploadtus.cpp

```diff
@@ -78,15 +78,19 @@ UploadDevice *PropagateUploadFileTUS::prepareDevice(const quint64 &chunkSize)
 SimpleNetworkJob *PropagateUploadFileTUS::makeCreationWithUploadJob(QNetworkRequest *request, UploadDevice *device)
 {
     Q_ASSERT(propagator()->account()->capabilities().tusSupport().extensions.contains(QStringLiteral("creation-with-upload")));
-    // in difference to the old protocol the algrithm and the value are space seperated
+
     const auto checksumHeader = ChecksumHeader::parseChecksumHeader(_transmissionChecksumHeader);
-    const QByteArray checkSum = QByteArray(CheckSums::toQString(checksumHeader.type()).toUtf8() + ' ' + checksumHeader.checksum()).toBase64();
-    const QByteArray base64Path = propagator()->fullRemotePath(_item->_file).toUtf8().toBase64();
-    qCDebug(lcPropagateUploadTUS) << "FullPath:" << propagator()->fullRemotePath(_item->_file) << "Base64:" << base64Path;
-    request->setRawHeader(QByteArrayLiteral("Upload-Metadata"), "filename " + base64Path + ",checksum " + checkSum);
+
+    QByteArrayList encodedMetaData;
+    auto addMetaData = [&encodedMetaData](const QByteArray &key, const QByteArray &value) { encodedMetaData << key + ' ' + value.toBase64(); };
```

Comment:
um, doesn't this result in `filename foo.txtchecksum sha1`? So does addMetaDatta need to add line separators or something? Or does the `<<` operator do that?

Reply:
Its a list QByteArrayList which later gets joined
`key base64Val`

Reply:
Doh you're right. My bad.